pkg/git/libgit2/managed/http.go

```diff
@@ -362,21 +364,31 @@ func (self *httpSmartSubtransportStream) sendRequest() error {
 		// GET requests will be automatically redirected.
 		// POST require the new destination, and also the body content.
 		if req.Method == "POST" && resp.StatusCode >= 301 && resp.StatusCode <= 308 {
+			// ensure body is fully processed and closed
+			// for increased likelihood of transport reuse in HTTP/1.x.
+			io.Copy(io.Discard, resp.Body)
```

Comment:
Some error check would be nice, just in case.
And for cases where we can't return error, like the `Free()` above, maybe we could make use of the trace logger to log in case of any error while copying. Maybe we could do the same for all the response body close calls too.

Reply:
I reviewed places that errors were not properly handled. Some of which can be safely ignored, as an error would not really be meaningful, and added a comment to that effect. Also added error handling on the `.Close()` instances that were appropriate.